sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpBodyType.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Messaging.ServiceBus.Amqp
+{
+    /// <summary>
```

Comment:
We should probably consider splitting the general bits (not Service Bus specific) into an area of Azure.Core, either as a shared code asset or part of the public building blocks.

Event Hubs may have a slightly different implementation due to maintaining backwards compatibility, but I can see this enum, the transport body, and some of the other bits being used as-is.

Reply:
That makes sense to me. How about, when we want other services like EH to use this, we move it. We'd want to do this prior to SB GA, though. But, to get this into a preview, it existing in just SB should be fine, I believe.

Reply:
Works for me!

Reply:
We could put this in Azure.Core.Experimental for now, so it will get some time to bake before trying to put in Azure.Core.